themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,548 @@
+
+Let's see how. Here's a revised architecture diagram showing how you might approach building an application like this one with EventBridge positioned between API Gateway and Lambda:
+
+![An expanded diagram showing API Gateway now publishing to EventBridge, with EventBridge invoking two separate Lambda functions: one writing an order to a database, the other posting a message to Slack.](./figure-3.png)
+
+Now let's have a look at what it'd be like to build it with Pulumi. We won't build _everything_ in this diagram---things like writing to the database or messaging Slack are left for you to explore---but we will build enough to give you clear picture and a working example of how to connect all of these parts into a working application. Specifically:
+
+* an API Gateway [instance](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-vs-rest.html) to act as a container for your public API, along with a [_stage_](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-stages.html) and a [_route_](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-routes.html) to handle inbound HTTP requests;
+* an EventBridge [integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations.html) (comprised of an event bus and event rule) to handle notifications from API Gateway; and finally,
+* one or more Lambda functions to be invoked in response to event-rule matches.
+
+Let's get started.
+
+## Create a new project and stack
+
+As always, it’s good practice to begin with a new project and stack:
+
+{{< chooser language "typescript,python" />}}
+
+{{% choosable language typescript %}}
+
+```bash
+$ pulumi new aws-typescript
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```bash
+$ pulumi new aws-python
+```
+
+{{% /choosable %}}
+
+Make sure you [configure your AWS credentials]({{< relref "/registry/packages/aws/installation-configuration" >}}) as well, and when prompted, choose whatever stack name and AWS region work best for you.
+
+{{% notes %}}
+
+In this post, we'll be using [API Gateway V2](https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/api-reference.html) as it's newer and has a few features (like [auto-deploy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-stage.html#aws-resource-apigatewayv2-stage-properties)) that make it easier to work with in this context. The same ideas apply to both V1 and V2, though, so while V2 still lacks full feature parity with V1, the underlying resources we'll be using---methods, routes, integrations---are the same. You'll find links to examples of both at the end of this post.
+
+{{% /notes %}}
+
+## Create the API gateway and stage
+
+Start by replacing the contents of {{% langfile %}} with the following code to declare a new API Gateway [API]({{< relref "/registry/packages/aws/api-docs/apigatewayv2/api" >}}):
```

Comment:
Not sure why, but this link isn't pointing to the right spot in the preview (it's showing http://pulumi-hugo-origin-pr-955-01b72936.s3-website.us-west-2.amazonaws.com/blog/api-gateway-to-eventbridge/)
![Screen Shot 2022-01-11 at 4 17 33 PM](https://user-images.githubusercontent.com/1538692/149030375-bd6f58d3-edaf-4f59-b7fc-999bb493d3b1.png)

Reply:
Ah yeah, this is just a side-effect of not having the registry built into the preview. When we merge it, it'll point to https://pulumi.com/registry/packages/aws/api-docs/apigatewayv2/api. (There's a lil' note about this weirdness here: https://github.com/pulumi/pulumi-hugo#about-this-repository)

Reply:
Huh; I've at least had the link render properly but 404 (or, at least, I think I have? Maybe that's been on local 😅 )